### DISCUSSION TITLE
No access to web server on the Raspberry Pi Pico W from external networks

### DISCUSSION THREAD

**Discussion by CHPeter** [micropython/micropython#11934] [RP2040 / Pico]
Hello
WEB server Raspberry Pi Pico W is ideally accessible from a local network and does not respond to a request from an external network - for example http://217.65.242.xx/: 34911.
Port 34911 on the router translates to port 80.
If I register port 80 in Firewall NAT instead of port 34911, everything works.
Please help me.
I use:
MicroPython v1.20.0 2023-04-26
Last firmware from https://rpf.io/pico-w-firmware.

Thank you.
Peter.

[pico_w_network.docx](https://github.com/micropython/micropython/files/11938079/pico_w_network.docx)

**Comment by bidrohini** [comment]
Verify that your router's firewall is not blocking incoming connections on port 34911 or port 80. If necessary, temporarily disable the firewall to test if it is causing the issue. Ensure that you have appropriate security measures in place before disabling the firewall.

**Comment by karfas** [comment]
I second @bidrohini. This looks like some firewall/network problem, not like a micropython issue, especially because "everything works" with port 80.
Either your port rewriting (34911 => 80) isn't working (unlikely) or *something* along the way from your test client to the Pico might block port 34911. Try another port, check the firewalls of all involved providers, ...

**Reply by CHPeter** [reply]
Thank you very much for the advice, but I already did it and it did not help.

**Reply by bidrohini** [reply]
There may be some other issues too. Some ISPs or network providers impose additional restrictions that may prevent external access to certain ports. Contact your ISP or network administrator to inquire if there are any such restrictions in place. Also, make sure that you have set up port forwarding correctly to forward incoming requests on port 34911 to the internal IP address of your Raspberry Pi Pico W on port 80.

**Comment by GM-Script-Writer-62850** [comment]
Some ISPs block port 80 incoming, try using 81 externally
